fraqbot/Local/randomemoji.py

```diff
@@ -0,0 +1,63 @@
+import logging
+import os
+import random
+import sys
+
+from Legobot.Lego import Lego
+
+
+logger = logging.getLogger(__name__)
+LOCAL_DIR = os.path.join(os.path.abspath(os.path.dirname(__file__)))
+
+if LOCAL_DIR not in sys.path:
+    sys.path.append(LOCAL_DIR)
+
+
+from helpers import utils  # noqa: E402
+
+
+class RandomEmoji(Lego):
+    def __init__(self, baseplate, lock, *args, **kwargs):
+        super().__init__(baseplate, lock, acl=kwargs.get('acl'))
+
+    def listening_for(self, message):
+        text = message.get('text')
+
+        return isinstance(text, str) and text.startswith('!emoji')
+
+    def _get_emoji(self, how_many):
+        maximum_how_many = 20
+        minimum_how_many = 1
+        how_many_limited = max(minimum_how_many, min(maximum_how_many, how_many))
+        
+        emoji_response = utils.call_slack_api(
+                self.slack_client,
```

Comment:
slack_client is not a native property of our Legos, but the client can be derived from the pykka system.
See https://github.com/ReformedDevs/fraqbot/blob/master/fraqbot/Local/helpers/coins.py#L25
and https://github.com/ReformedDevs/fraqbot/blob/master/fraqbot/Local/helpers/coins.py#L73
and https://github.com/ReformedDevs/fraqbot/blob/5fd6a93b445ba288d80f8cdb618d1273b8ead1d1/fraqbot/Local/coins.py#L221

Since you're not using the botThread for anything else, you could just save its client as this Lego's client property. Be sure to set a default if no bothThread is found in the pikka system (`self.client = None`)